l1-contracts/deploy-scripts/LensScript.s.sol

```diff
@@ -0,0 +1,292 @@
+            })
+        );
+        
+        tempEcoExecuteOperation = abi.encodeWithSelector(
+            IGovernance.execute.selector,
+            IGovernance.Operation({
+                calls: tempEcoOwnershipCalls,
+                predecessor: bytes32(0),
+                salt: bytes32(0)
+            })
+        );
+    }
+
+    function _generateAcceptOwnershipCalls() internal {
+        // bh
+        calls.push(
+            IGovernance.Call({
+                target: config.tempBridgehub,
+                value: 0,
+                data: abi.encodeCall(Ownable2Step.acceptOwnership, ())
+            })
+        );
+
+        // stm
+        calls.push(
+            IGovernance.Call({
+                target: config.tempStateTransitionManager,
+                value: 0,
+                data: abi.encodeCall(Ownable2Step.acceptOwnership, ())
+            })
+        );
+
+        // vt
+        calls.push(
+            IGovernance.Call({
+                target: config.tempValidatorTimelock,
+                value: 0,
+                data: abi.encodeCall(Ownable2Step.acceptOwnership, ())
+            })
+        );
+    }
+
+    function _generateRegisterChainCall() internal {
+        calls.push(
+            IGovernance.Call({
+                target: config.newStateTransitionManager,
+                value: 0,
+                data: abi.encodeCall(
+                    IStateTransitionManager.registerAlreadyDeployedHyperchain,
+                    (config.lensChainId, config.lensDiamondProxy)
+                )
+            })
+        );
+    }
+
+    function _generateCreateChainCall() internal {
+        calls.push(
+            IGovernance.Call({
+                target: config.newBridgehub,
+                value: 0,
+                data: abi.encodeCall(
+                    IBridgehub.createNewChain,
+                    (
+                        config.lensChainId,
+                        config.lensDiamondProxy,
+                        config.baseToken,
+                        uint256(0),
+                        address(0),
+                        "0x"
+                    )
+                )
+            })
+        );
+    }
+
+    function _generateExecuteUpgradeCall() internal {
+        Diamond.FacetCut[] memory facetCuts = new Diamond.FacetCut[](0);
+
+        MigrationParams memory migrationParams = MigrationParams({
+            newVerifier: config.newVerifier,
+            newCTM: config.newStateTransitionManager,
+            newBridgehub: config.newBridgehub
+        });
+
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: migrationUpgradeAddress,
+            initCalldata: abi.encode(migrationParams)
+        });
+
+        calls.push(
+            IGovernance.Call({
+                target: config.lensDiamondProxy,
+                value: 0,
+                data: abi.encodeCall(IAdmin.executeUpgrade, (diamondCut))
```

Comment:
wont work, only CTM can call this function, you should invoke the CTM's function that will call this method